spec/controllers/openid_connect/authorization_controller_spec.rb

```diff
@@ -152,21 +152,73 @@
+            context 'user has a pending profile' do
+              context 'user has a gpo pending profile' do
+                let(:user) { create(:profile, :verify_by_mail_pending).user }
+
+                it 'redirects to gpo verify page' do
+                  action
+                  expect(controller).to redirect_to(idv_gpo_verify_url)
+                end
+              end
 
-              it 'redirects to fraud review page if fraud review is pending' do
-                action
-                expect(controller).to redirect_to(idv_please_call_url)
+              context 'user has an in person pending profile' do
+                let(:user) { create(:profile, :in_person_verification_pending).user }
+
+                it 'redirects to gpo verify page' do
+                  action
+                  expect(controller).to redirect_to(idv_in_person_ready_to_verify_url)
+                end
               end
-            end
 
-            context 'user is rejected due to fraud' do
-              let(:user) { create(:profile, :fraud_rejection).user }
+              context 'user is under fraud review' do
+                let(:user) { create(:profile, :fraud_review_pending).user }
 
-              it 'redirects to fraud rejection page if user is fraud rejected ' do
-                action
-                expect(controller).to redirect_to(idv_not_verified_url)
+                it 'redirects to fraud review page if fraud review is pending' do
+                  action
+                  expect(controller).to redirect_to(idv_please_call_url)
+                end
+              end
+
+              context 'user is rejected due to fraud' do
+                let(:user) { create(:profile, :fraud_rejection).user }
+
+                it 'redirects to fraud rejection page if user is fraud rejected ' do
+                  action
+                  expect(controller).to redirect_to(idv_not_verified_url)
+                end
+              end
+
+              context 'user has two pending reasons' do
+                context 'user has gpo and fraud review pending' do
+                  let(:user) do
+                    create(
+                      :profile,
+                      :verify_by_mail_pending,
+                      :fraud_review_pending,
+                    ).user
+                  end
+
+                  it 'redirects to gpo verify page if user has gpo and fraud pending' do
+                    action
+                    expect(controller).to redirect_to(idv_gpo_verify_url)
+                  end
+                end
+
+                context 'user has gpo and in person pending' do
+                  let(:user) do
+                    create(
+                      :profile,
+                      :verify_by_mail_pending,
+                      :in_person_verification_pending,
+                    ).user
+                  end
+
+                  it 'redirects to gpo verify page if user has gpo and in person pending' do
```

Comment:
Same suggestion as above.
```suggestion
                  it 'redirects to gpo verify page' do
```